stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -0,0 +1,4399 @@
+use std::collections::{HashMap, HashSet, VecDeque};
```

Comment:
Per our meeting earlier, I think the only things that this file needs to test are:

* PoX 4 instantiates 
* STX locked in PoX 3 unlock automatically
* Once PoX 4 is active, only PoX 4 stacking methods work.  PoX 3 ones fail.

There's no reason to test PoX 2 and PoX behavior, since both these contracts have been defunct for some time.

Reply:
@jcnelson
>* PoX 4 instantiates

does [`get_pox_addrs`](https://github.com/stacks-network/stacks-core/pull/4106/files#diff-1b7e3da4025244bdde5f977c36554b94ab89e844c05e4c3c0a106dd9da9c7636R3369) captures this? If it does, I would like to start the current `pox_4_tests.rs` (in this PR) from a clean state and start by just adding this test, and then keep adding to it.